extensions/ui/org.eclipse.smarthome.ui.basic/web-src/smarthome.js

```diff
@@ -1755,15 +1774,32 @@
 
 					var
 						item = widget.item.name,
-						value = widget.item.state,
+						state = widget.item.state,
 						label = widget.label,
+						value,
 						labelcolor = widget.labelcolor,
 						valuecolor = widget.valuecolor;
 
+					if (
+						(typeof(label) === "string") &&
+						(label.indexOf("[") !== -1) &&
+						(label.indexOf("]") !== -1)
+					) {
+						var
+							pos = label.indexOf("[");
+
+						value = label.substr(
+							pos + 1,
+							label.lastIndexOf("]") - (pos + 1)
+						);
```

Comment:
Since value extraction code is now repeated 3 times (2 here and 1 in SSE listener) I think it makes sense to extract it in a function which would take formatted string `Temperature [23 °C]` and return either a value `23 °C` or a string itself. AbstractChangeListener is probably a good place to put it in.